panel/backends/wayland/kwin_wayland/lxqtwmbackend_kwinwayland.cpp

```diff
@@ -0,0 +1,805 @@
+    // });
+
+    // Handle transient.
+    if (LXQtTaskBarPlasmaWindow *leader = window->parentWindow.data())
+    {
+        transients.insert(window, leader);
+
+        // Update demands attention state for leader.
+        if (window->windowState.testFlag(LXQtTaskBarPlasmaWindow::state::state_demands_attention))
+        {
+            transientsDemandingAttention.insert(leader, window);
+            if(leader->acceptedInTaskBar)
+                emit windowPropertyChanged(leader->getWindowId(), int(LXQtTaskBarWindowProperty::Urgency));
+        }
+    }
+    else
+    {
+        windows.emplace_back(window);
+        updateWindowAcceptance(window);
+    }
+}
+
+bool LXQtWMBackend_KWinWayland::acceptWindow(LXQtTaskBarPlasmaWindow *window) const
+{
+    if(window->windowState.testFlag(LXQtTaskBarPlasmaWindow::state::state_skiptaskbar))
+        return false;
+
+    if(transients.contains(window))
+        return false;
+
+    return true;
+}
+
+void LXQtWMBackend_KWinWayland::updateWindowAcceptance(LXQtTaskBarPlasmaWindow *window)
+{
+    if(!window->acceptedInTaskBar && acceptWindow(window))
+    {
+        window->acceptedInTaskBar = true;
+        emit windowAdded(window->getWindowId());
+    }
+    else if(window->acceptedInTaskBar && !acceptWindow(window))
+    {
+        window->acceptedInTaskBar = false;
+        emit windowRemoved(window->getWindowId());
+    }
+}
+
+LXQtTaskBarPlasmaWindow *LXQtWMBackend_KWinWayland::getWindow(WId windowId) const
+{
+    for(auto &window : std::as_const(windows))
+    {
+        if(window->getWindowId() == windowId)
+            return window.get();
+    }
+
+    return nullptr;
+}
+
+static inline QByteArray detectDesktopEnvironment()
+{
+    const QByteArray xdgCurrentDesktop = qgetenv("XDG_CURRENT_DESKTOP");
+    if (!xdgCurrentDesktop.isEmpty())
+        return xdgCurrentDesktop.toUpper(); // KDE, GNOME, UNITY, LXDE, MATE, XFCE...
+
+    // Classic fallbacks
+    if (!qEnvironmentVariableIsEmpty("KDE_FULL_SESSION"))
+        return QByteArrayLiteral("KDE");
+
+    // Fallback to checking $DESKTOP_SESSION (unreliable)
+    QByteArray desktopSession = qgetenv("DESKTOP_SESSION");
+
+    // This can be a path in /usr/share/xsessions
+    int slash = desktopSession.lastIndexOf('/');
+    // try decoding just the basename
+    desktopSession = desktopSession.mid(slash + 1);
+
+    if (desktopSession == "kde" || desktopSession == "plasma")
+        return QByteArrayLiteral("KDE");
+
+    return QByteArray();
+}
+
+int LXQtWMBackendKWinWaylandLibrary::getBackendScore() const
+{
+    auto *waylandApplication = qGuiApp->nativeInterface<QNativeInterface::QWaylandApplication>();
+    if(!waylandApplication)
+        return 0;
+
+    // Detect KWin Plasma
+    QByteArray desktop = detectDesktopEnvironment();
+    if(desktop == "KDE")
+        return 100;
+
+    // It's still somewhat useful for other wayland compositors
```

Comment:
Please correct me if I am wrong, but I think kwin/plasma would not be of any help on wlroots based compositors, what-so-ever.

Reply:
Done!